quickwit/quickwit-indexing/src/actors/indexing_service.rs

```diff
@@ -96,6 +105,12 @@ pub struct IndexingService {
     pipeline_handles: HashMap<IndexingPipelineId, ActorHandle<IndexingPipeline>>,
     state: IndexingServiceState,
     enable_ingest_api: bool,
+    // Hashmap of of index_id to merge policy.
+    merge_policies: HashMap<String, Arc<dyn MergePolicy>>,
+    // Hashmap of (index_id, source_id) to indexing directory.
+    indexing_directories: HashMap<String, IndexingDirectory>,
```

Comment:
@guilload I preferred a combination a  String to avoid 2x allocations, besides constructs like `(pipeline_id.index_id.clone(), pipeline.source_id.clone())` would have been creeping.